include/qx/Instructions.hpp

```diff
@@ -26,8 +26,8 @@ struct ControlledInstruction : public Instruction {
     ControlBits control_bits;
     std::shared_ptr<Instruction> instruction;
 
-    ~ControlledInstruction() = default;
-    ControlledInstruction(ControlBits const& control_bits, std::shared_ptr<Instruction> instruction);
+    ~ControlledInstruction() override = default;
```

Comment:
Just for my understanding: What does `override` do?

Reply:
It just adds information. It's not mandatory. But it tells you that that method is part of a chain of virtaul methods, i.e., there is a base class where that method is defined. So you are just overriding it, giving it a more specific implementation.